blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/PipelineImpl.java

```diff
@@ -179,4 +179,9 @@ public Resource resolve(Item context, Reachable parent, Item target) {
 
     }
 
+    @Exported(inline = true)
+    @Navigable
+    public Container<Resource> getActivities() {
+        return Containers.fromResource(getLink(),Lists.newArrayList(Iterators.concat(getQueue().iterator(), getRuns().iterator())));
```

Comment:
👍🏻

@vivek oh just past experience with things like hibernate that hide away the fetching logic from you. You trip up badly and you load an entire table into memory - but if you are happy I am happy ;)